backend/src/middleware/validation/index.js

```diff
@@ -0,0 +1,31 @@
+import { UserInputError } from 'apollo-server'
+
+const USERNAME_MIN_LENGTH = 3
+
+const validateUsername = async (resolve, root, args, context, info) => {
+  if (args.name && args.name.length >= USERNAME_MIN_LENGTH) {
+    /* eslint-disable-next-line no-return-await */
+    return await resolve(root, args, context, info)
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-05-10T17:18:54Z" title="Friday, May 10th 2019, 7:18:54 pm +02:00">May 10, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,31 @@
+import { UserInputError } from 'apollo-server'
+
+const USERNAME_MIN_LENGTH = 3
+
+const validateUsername = async (resolve, root, args, context, info) => {
+  if (args.name && args.name.length >= USERNAME_MIN_LENGTH) {
+    /* eslint-disable-next-line no-return-await */
+    return await resolve(root, args, context, info)
```


are we updating the user twice here? once in the validation middleware and once on line 15 of the `userMiddleware`?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-10T17:58:02Z" title="Friday, May 10th 2019, 7:58:02 pm +02:00">May 10, 2019</time>_

---


No, calling `resolve` will continue the chain of middlewares until you hit the final resolver.